### PR TITLE
Add TFLite benchmark workflow

### PR DESCRIPTION
**Pull request by mariecwhite**

### CODE REVIEW

File: experimental/tflite/benchmark_tflite_android.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+# Setup benchmarking environment.
+adb shell "rm -rf ${ROOT_DIR}"
+adb shell "mkdir ${ROOT_DIR}"
+
+adb push "${TD}/../../comparative_benchmark/scripts/set_android_scaling_governor.sh" "${ROOT_DIR}"
+adb shell "chmod +x ${ROOT_DIR}/set_android_scaling_governor.sh"
+adb shell "su root sh ${ROOT_DIR}/set_android_scaling_governor.sh performance"
+#adb shell "su root sendhint -m DISPLAY_INACTIVE -e 0"
+adb shell "su root setprop persist.vendor.disable.thermal.control 1"
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply experimental/tflite/benchmark_tflite_android.sh:37]: This is a vendor property and probably only applies to Pixel phones. I think it makes more sense to keep them separate.

File: .github/workflows/run_mobile_comparative_benchmark.yml [hunk 2]
```diff
@@ -250,9 +275,12 @@ jobs:
       BENCHMARK_GCS_DIR: ${{ needs.setup.outputs.benchmark-gcs-dir }}
       RESULTS_DIR: results-dir
       TARGET_DEVICE: pixel-6-pro
-      IREE_ANDROID_TOOLS_DIR: ${{ needs.build_iree.outputs.iree-android-tools-dir }}
-      IREE_ANDROID_TOOLS_DIR_ARCHIVE: ${{ needs.build_iree.outputs.iree-android-tools-dir-archive }}
-      IREE_ANDROID_TOOLS_DIR_GCS_ARTIFACT: ${{ needs.build_iree.outputs.iree-android-tools-dir-gcs-artifact }}
+      # IREE_ANDROID_TOOLS_DIR: ${{ needs.build_iree.outputs.iree-android-tools-dir }}
+      # IREE_ANDROID_TOOLS_DIR_ARCHIVE: ${{ needs.build_iree.outputs.iree-android-tools-dir-archive }}
+      # IREE_ANDROID_TOOLS_DIR_GCS_ARTIFACT: ${{ needs.build_iree.outputs.iree-android-tools-dir-gcs-artifact }}
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply .github/workflows/run_mobile_comparative_benchmark.yml:280]: Oh yes, I hardcoded this to speed up testing. Cleaned up.

File: .github/workflows/run_mobile_comparative_benchmark.yml [hunk 3]
```diff
@@ -58,75 +59,76 @@ jobs:
...
-          IREE_INSTALL_DIR_GCS_ARTIFACT: ${{ env.BENCHMARK_GCS_DIR }}/${{ steps.archive.outputs.iree-install-dir-archive }}
-          IREE_ANDROID_TOOLS_DIR_ARCHIVE: ${{ steps.archive.outputs.iree-android-tools-dir-archive }}
-          IREE_ANDROID_TOOLS_DIR_GCS_ARTIFACT: ${{ env.BENCHMARK_GCS_DIR }}/${{ steps.archive.outputs.iree-android-tools-dir-archive }}
-        run: |
-          gcloud storage cp "${IREE_INSTALL_DIR_ARCHIVE}" "${IREE_INSTALL_DIR_GCS_ARTIFACT}"
-          echo "iree-install-dir-gcs-artifact=${IREE_INSTALL_DIR_GCS_ARTIFACT}" >> "${GITHUB_OUTPUT}"
-
-          gcloud storage cp "${IREE_ANDROID_TOOLS_DIR_ARCHIVE}" "${IREE_ANDROID_TOOLS_DIR_GCS_ARTIFACT}"
-          echo "iree-android-tools-dir-gcs-artifact=${IREE_ANDROID_TOOLS_DIR_GCS_ARTIFACT}" >> "${GITHUB_OUTPUT}"
+#  build_iree:
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply .github/workflows/run_mobile_comparative_benchmark.yml]: Reverted

**Review by pzread — APPROVED** [review]

File: experimental/tflite/benchmark_tflite_android.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  "models/BERT_BASE_FP16_TFLITE_I32_SEQLEN.+/.+"
+  "models/BERT_BASE_DYN_QUANT_TFLITE_I32_SEQLEN.+/.+"
+  "models/BERT_BASE_INT8_TFLITE_I32_SEQLEN.+/.+"
+  "models/VIT_CLASSIFICATION_FP32_TFLITE_3X224X224XF32/.+"
+  "models/VIT_CLASSIFICATION_FP16_TFLITE_3X224X224XF32/.+"
+  "models/VIT_CLASSIFICATION_DYN_QUANT_TFLITE_3X224X224XF32/.+"
+  "models/VIT_CLASSIFICATION_INT8_TFLITE_3X224X224XINT8/.+"
+)
+
+THREAD_CONFIG="{1: '80', 4: 'F0'}"
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply experimental/tflite/benchmark_tflite_android.sh:64]: Added

File: experimental/tflite/benchmark_tflite_x86.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  "models/BERT_BASE_FP16_TFLITE_I32_SEQLEN.+/.+"
+  "models/BERT_BASE_DYN_QUANT_TFLITE_I32_SEQLEN.+/.+"
+  "models/BERT_BASE_INT8_TFLITE_I32_SEQLEN.+/.+"
+  "models/VIT_CLASSIFICATION_FP32_TFLITE_3X224X224XF32/.+"
+  "models/VIT_CLASSIFICATION_FP16_TFLITE_3X224X224XF32/.+"
+  "models/VIT_CLASSIFICATION_DYN_QUANT_TFLITE_3X224X224XF32/.+"
+  "models/VIT_CLASSIFICATION_INT8_TFLITE_3X224X224XINT8/.+"
+)
+
+THREAD_CONFIG="{1: '', 8: '', 15: '', 30: ''}"
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply experimental/tflite/benchmark_tflite_x86.sh:48]: Added.

**Pull request merged**